src/main/java/com/guciowons/shoppingbasket/Basket/BasketService.java

```diff
@@ -5,31 +5,46 @@
 
 @Service
 public class BasketService {
-    private Basket basket = new Basket();
-
     private final ProductDao productDao;
     private final BasketSummarizer basketSummarizer;
+    private final BasketDao basketDao;
 
-    public BasketService(ProductDao productDao, BasketSummarizer basketSummarizer) {
+    public BasketService(ProductDao productDao, BasketSummarizer basketSummarizer, BasketDao basketDao) {
         this.productDao = productDao;
         this.basketSummarizer = basketSummarizer;
+        this.basketDao = basketDao;
     }
 
-    public void addProductToBasket(int id, int quantity) throws IllegalArgumentException{
-        productDao.findById(id).ifPresentOrElse(
-                product -> basket.addProduct(id, quantity),
-                () -> {throw new IllegalArgumentException("No such product");}
-        );
+    public Basket createBasket() {
+        Basket newBasket = new Basket(basketDao.getAll().size());
```

Comment:
To jest okej TYLKO w przypadku, kiedy nie masz usuwania koszyków.
Jak dojdzie usuwanie to tutaj będzie występował problem koszyków z tym samym ID.

ID powinno być najlepiej jakąś losową liczbą/ciągiem znaków -- to powoduje najmniej problemów. Każde inne rozwiązanie prędzej czy później coś psuje.

Ale na teraz jest okej.